ament_cpplint/ament_cpplint/main.py

```diff
@@ -144,27 +148,29 @@ def main(argv=sys.argv[1:]):
         else:
             print("Not using '--root'")
         print('')
-        arguments += files
-        filenames = ParseArguments(arguments)
-
-        for filename in filenames:
-            # hook into error reporting
-            errors = []
-
-            def custom_error(filename, linenum, category, confidence, message):
-                if ament_cpplint.cpplint._ShouldPrintError(category, confidence, linenum):
-                    errors.append({
-                        'linenum': linenum,
-                        'category': category,
-                        'confidence': confidence,
-                        'message': message,
-                    })
-                DefaultError(filename, linenum, category, confidence, message)
-            ament_cpplint.cpplint.Error = custom_error
-
-            ProcessFile(filename, _cpplint_state.verbose_level)
-            report.append((filename, errors))
-            print('')
+        files_check = [f for f in files if not any(f for excl in args.excludes if excl in f)]
+        if(len(files_check) != 0):
```

Comment:
@dirk-thomas  I agree with your comment https://github.com/ament/ament_lint/pull/238#discussion_r436100214, but I added this condition to avoid this case

Reply:
How is this condition going to help the problem that the *wrong* files are being excluded in the line above?

Reply:
your example is a valid case, it's when you want to exclude a hole folder.

files = ['foo/bar/baz']
args.excludes = ['bar']

files_check in this case is empty, The condition allows us to avoid running this test

Reply:
No, in that example the directory `foo/bar/baz` should *not* be excluded since the excludes only list the name `bar` which should only match a file / directory which is a sibling of `foo`.

To make the example even more obvious:

```
files = ['foobarbaz']
args.excludes = ['bar']
```

Your logic does exclude the file / directory `foobarbaz` which is wrong.

Reply:
I modified the logic https://github.com/ament/ament_lint/pull/238/commits/66ed86bd3baf3ebd942572ae0adefc4ea21b3f93 

I get the file_name and tokenize the path. Then if one of the excludes match with file_name or the path_token I add this file to an avoid file list. Finally I remove the common elements to get only the files to test.

Reply:
I included pathlib to check with all the parent if the there are any matches https://github.com/ament/ament_lint/pull/238/commits/31ea18d59ed3ad0e359bdc64511c6a670d8ac24f